<a href="https://colab.research.google.com/github/vsauldba/python-data-analytics/blob/main/hw2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [37]:
# La Guardia Community College
#Course: Python
#Student's name: Vilio Saul
#Subject: Analyzing Vehicle Gas Mileage Data Using Pandas.
#Prof.: Imran Khan
#Date: 9/2/2026


In [38]:
# 1-Import The Pandas library
import pandas as pd

In [39]:
# 2-Load the mtcars dataset from Github
df = pd.read_csv('https://raw.githubusercontent.com/CunyLaguardiaDataAnalytics/datasets/refs/heads/master/mtcars.csv')

In [40]:
# 3-Displays the first 5 rows of the dataset
df.head()

,Unnamed: 0,mpg,cyl,disp,hp,drat,wt,qsec,vs,am,gear,carb
0,Mazda RX4,21.0,6,160.0,110,3.90,2.620,16.46,0,1,4,4
1,Mazda RX4 Wag,21.0,6,160.0,110,3.90,2.875,17.02,0,1,4,4
2,Datsun 710,22.8,4,108.0,93,3.85,2.320,18.61,1,1,4,1
3,Hornet 4 Drive,21.4,6,258.0,110,3.08,3.215,19.44,1,0,3,1
4,Hornet Sportabout,18.7,8,360.0,175,3.15,3.440,17.02,0,0,3,2


In [41]:
# 4-Display the  number of rows and columns
df.shape

(32, 12)

In [42]:
# 5-Display all Column Names
df.columns


Index(['Unnamed: 0', 'mpg', 'cyl', 'disp', 'hp', 'drat', 'wt', 'qsec', 'vs',
       'am', 'gear', 'carb'],
      dtype='object')

In [43]:
# 6-Get Information about the dataset such as: Column Names, Data Types and Missing Values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 12 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  32 non-null     object 
 1   mpg         32 non-null     float64
 2   cyl         32 non-null     int64  
 3   disp        32 non-null     float64
 4   hp          32 non-null     int64  
 5   drat        32 non-null     float64
 6   wt          32 non-null     float64
 7   qsec        32 non-null     float64
 8   vs          32 non-null     int64  
 9   am          32 non-null     int64  
 10  gear        32 non-null     int64  
 11  carb        32 non-null     int64  
dtypes: float64(5), int64(6), object(1)
memory usage: 3.1+ KB


In [44]:
# 7-Display Summary Statistics for the numeric Columns
df.describe()

,mpg,cyl,disp,hp,drat,wt,qsec,vs,am,gear,carb
count,32.000000,32.000000,32.000000,32.000000,32.000000,32.000000,32.000000,32.000000,32.000000,32.000000,32.0000
mean,20.090625,6.187500,230.721875,146.687500,3.596563,3.217250,17.848750,0.437500,0.406250,3.687500,2.8125
std,6.026948,1.785922,123.938694,68.562868,0.534679,0.978457,1.786943,0.504016,0.498991,0.737804,1.6152
min,10.400000,4.000000,71.100000,52.000000,2.760000,1.513000,14.500000,0.000000,0.000000,3.000000,1.0000
25%,15.425000,4.000000,120.825000,96.500000,3.080000,2.581250,16.892500,0.000000,0.000000,3.000000,2.0000
50%,19.200000,6.000000,196.300000,123.000000,3.695000,3.325000,17.710000,0.000000,0.000000,4.000000,2.0000
75%,22.800000,8.000000,326.000000,180.000000,3.920000,3.610000,18.900000,1.000000,1.000000,4.000000,4.0000
max,33.900000,8.000000,472.000000,335.000000,4.930000,5.424000,22.900000,1.000000,1.000000,5.000000,8.0000


In [45]:
# 8-Rename the missing Columns
df.rename(columns={'Unnamed: 0':'car'}, inplace=True)

In [46]:
# 9-Displays the 5 rows after renaming the column
df.head()

,car,mpg,cyl,disp,hp,drat,wt,qsec,vs,am,gear,carb
0,Mazda RX4,21.0,6,160.0,110,3.90,2.620,16.46,0,1,4,4
1,Mazda RX4 Wag,21.0,6,160.0,110,3.90,2.875,17.02,0,1,4,4
2,Datsun 710,22.8,4,108.0,93,3.85,2.320,18.61,1,1,4,1
3,Hornet 4 Drive,21.4,6,258.0,110,3.08,3.215,19.44,1,0,3,1
4,Hornet Sportabout,18.7,8,360.0,175,3.15,3.440,17.02,0,0,3,2


In [47]:
# 10- Create a new dataframe with only the required columns such as: mpg, hp, wt and cyl.
cars = df[['mpg', 'hp', 'wt', 'cyl']]
# Show the first 5 rows of the new dataframe
cars.head()

,mpg,hp,wt,cyl
0,21.0,110,2.620,6
1,21.0,110,2.875,6
2,22.8,93,2.320,4
3,21.4,110,3.215,6
4,18.7,175,3.440,8


In [48]:
# 10- Rename the "cyl" to "cylinders"
cars = cars.rename(columns={"cyl":"cylinders"})



In [49]:
# 11- Display the new column name
cars.head()

,mpg,hp,wt,cylinders
0,21.0,110,2.620,6
1,21.0,110,2.875,6
2,22.8,93,2.320,4
3,21.4,110,3.215,6
4,18.7,175,3.440,8


In [50]:
# 12- Displays Summary statistics for mpg, hp, wt and cylinders
cars.describe()

,mpg,hp,wt,cylinders
count,32.000000,32.000000,32.000000,32.000000
mean,20.090625,146.687500,3.217250,6.187500
std,6.026948,68.562868,0.978457,1.785922
min,10.400000,52.000000,1.513000,4.000000
25%,15.425000,96.500000,2.581250,4.000000
50%,19.200000,123.000000,3.325000,6.000000
75%,22.800000,180.000000,3.610000,8.000000
max,33.900000,335.000000,5.424000,8.000000


In [51]:
# 13-Calculate the average miles per gallon
cars['mpg'].mean()

np.float64(20.090625000000003)

Comments: The average MPG for the vehicle is approximately **20.09 miles per gallon.**


In [52]:
# 14-Select cars with more than 150 horsepower
high_hp = cars[cars['hp'] > 150]


In [53]:
# 15- Displays cars with more than 150 horsepower
high_hp

,mpg,hp,wt,cylinders
4,18.7,175,3.440,8
6,14.3,245,3.570,8
11,16.4,180,4.070,8
12,17.3,180,3.730,8
13,15.2,180,3.780,8
14,10.4,205,5.250,8
15,10.4,215,5.424,8
16,14.7,230,5.345,8
23,13.3,245,3.840,8
24,19.2,175,3.845,8


In [54]:
# 16- Display summary statistics for cars with more than 150horsepower
high_hp.describe()

,mpg,hp,wt,cylinders
count,13.000000,13.000000,13.000000,13.000000
mean,15.415385,215.692308,3.984923,7.846154
std,2.956588,47.779676,0.839512,0.554700
min,10.400000,175.000000,2.770000,6.000000
25%,14.300000,180.000000,3.570000,8.000000
50%,15.200000,205.000000,3.780000,8.000000
75%,17.300000,245.000000,4.070000,8.000000
max,19.700000,335.000000,5.424000,8.000000


In [55]:
# 17- Calculate the average horsepower
cars['hp'].mean()

np.float64(146.6875)

The average horsepower of the vehicle is approximately 146.69 hp.

Summary: I used Pandas to explore the mtcars dataset. I looked at MPG, horsepower, weight and cylinders. I calculated averages and used summary statistics to better understand the vehicle data.